hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -270,9 +270,12 @@ && equal( join.getKey(), manyToOne ) ) {
 				}
 				this.bidirectionalAttributeName = bidirectionalAttributeName;
 			}
-			else {
+			else if ( cardinality == Cardinality.MANY_TO_ONE ) {
 				this.bidirectionalAttributeName = referencedPropertyName;
 			}
+			else {
+				this.bidirectionalAttributeName = null;
+			}
```

Comment:
This change aims to correctly detect bi-directional circular associations when the attribute mapping is the `mappedBy` side of a `OneToOne` association with `FetchType.EAGER`. Without this, the circular fetch would not be recognized as bi-directional thus triggering an additional query identical to the first one which was not needed.

Setting `bidirectionalAttributeName` to null allows the method `isBidirectionalAttributeName` to check the parent mapping for bi-directionality obtaining a correct result.

Reply:
It feels a bit fishy to set `bidirectionalAttributeName` to null even though we know the value. Is it not possible to fix `isBidirectionalAttributeName` without this change?

Reply:
I did try that at first but I was only able to make the check work by setting the property to null. It looked like `bidirectionalAttributeName` *expected* a null value when checking the `mappedBy` side of a one-to-one association, I couldn't find any other condition that wouldn't break other test cases.

Reply:
Found another way to check bidirectionality without null, using `identifyingColumnsTableExpression`.